tests/kernel/sched/schedule_api/src/test_sched_priority.c

```diff
@@ -95,6 +111,63 @@ void test_priority_preemptible(void)
 	k_thread_priority_set(k_current_get(), old_prio);
 }
 
+/**
+ * @brief Validate scheduling sequence of preemptive threads with start delay
+ *
+ * @details Create three preemptive threads which are of priority
+ * higher than current thread. Make sure that the highest priority
+ * and longest waiting thread is scheduled first.
+ *
+ * @ingroup kernel_sched_tests
+ */
+void test_priority_preemptible_wait_prio(void)
+{
+	int old_prio = k_thread_priority_get(k_current_get());
+	k_tid_t tid[THREAD_NUM];
+	uint8_t tid_chk[4] =  { 0, 1, 2, 3 };
+
+	k_sem_init(&sync_sema, 0, THREAD_NUM);
+
+	/* set current thread to a non-negative priority */
+	last_prio = 2;
```

Comment:
Style: here, you're using a fixed number ("2") for a thread priority.  Below, you're using the (IMHO confusing -- preemptible and cooperative priorities count in different directions!) K_PRIO_* macros instead. 

Pick one convention.  I suggest this one.

Reply:
I will unify it.